### PR TITLE
io.nutils tool

### PR DESCRIPTION
**Pull request by LinusKnecht**
# new io.nutils

## Related issues
#7 relates to a new tool for gustaf - nutils conversions

## Description
This PR extends ```gustaf.io``` by a ```nutils``` tool. It can load and export simplex meshes (triangle and tetrahedron) from Gustaf to Nutils and vice versa. Functions can be called by

```
    load(fname)
    export(fname,mesh)
    to_nutils_simplex(mesh)
```
A ```io.mixd``` adaption was necessary. The new function ```as_mrng``` returns a mrng-like file.

### CODE REVIEW

File: gustaf/io/nutils.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    fname = abs_fname(fname)
+    check_and_makedirs(fname)
+
+    np.savez(fname, **dic)
+
+
+
+def to_nutils_simplex(mesh):
+    """
+    Converts a Gustaf_Mesh to a Dictionary, which can be interpreted by nutils.mesh.simplex(**to_nutils_simplex(mesh)). Only work for Triangles and Tetrahedrons!
```
Comment (j042) [inline gustaf/io/nutils.py:87]: 79 max line length.

File: gustaf/io/nutils.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+        volumes = mesh.volumes				
+        elements = volumes     
+    else:
+        raise TypeError('Only Triangle and Tetrahedrons are accepted.') 
+    
+    #Sort the Node IDs for each Element. 
+    elements_sorted = np.zeros(elements.shape)
+    sort_array = np.zeros(elements.shape)
+    
+    for index, row in enumerate(elements, start = 0):
```
Comment (j042) [inline gustaf/io/nutils.py:121]:
```suggestion
    for index, row in enumerate(elements):
```
isn't default value for start always 0?

**Review by j042 — CHANGES_REQUESTED** [review]
Thanks, nice feature!  
Let's try to stay close to [pep8](pep8.org) until we automate this process.

**Review by CodeFloh — COMMENTED** [review]
After we discussed the code I found a neat NumPy function that can replace some of your loops. This makes the code shorter and transfers some loops to NumPy.

Feel free to try this out or ignore it.

File: gustaf/io/nutils.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        
+    for bi in bound_id:
+        temp = []
+        for elid, el in enumerate(bcs_sorted):
+            for index, edge in enumerate(el):
+                if bi == edge:
+                    temp.append([elid,index])
+        bcs.update(
+            {str(bi)		:	np.array(temp)}
+        )
```
Comment (CodeFloh) [inline gustaf/io/nutils.py:153]:
```suggestion
        bcs.update(
            {str(bi) : np.argwhere(bcs_sorted == bi)}
        )
```
Shorter option.. ;)

**Comment by j042** [comment]
Closing this one for now, feel free to reopen! @LinusKnecht

File: gustaf/io/nutils.py [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+    
+    #Sort the Node IDs for each Element. 
+    elements_sorted = np.zeros(elements.shape)
+    sort_array = np.zeros(elements.shape)
+    
+    for index, row in enumerate(elements):
+        elements_sorted[index] = np.sort(row)	
+        sort_array[index] = np.argsort(row)
+    elements_sorted = elements_sorted.astype(int)	
+    sort_array = sort_array.astype(int)
```
[the remark this answers is not in the document]
Comment (LinusKnecht) [reply gustaf/io/nutils.py:126]: Thanks for the hints!

**Pull request closed without merging**